src/common/ant.ts

```diff
@@ -278,6 +280,92 @@ export class AoANTReadable implements AoANTRead {
     if (strict) parseSchemaResult(z.number(), balance);
     return balance;
   }
+
+  /**
+   * @returns {Promise<AoANTHandler[]>} The handlers of the ANT.
+   * @example
+   * Get the handlers of the ANT.
+   * ```ts
+   * const handlers = await ant.getHandlers();
+   * ```
+   */
+  async getHandlers(): Promise<AoANTHandler[]> {
+    const info = await this.getInfo();
+    const handlers = info.Handlers ?? info.HandlerNames;
+    if (
+      this.strict &&
+      !(
+        Array.isArray(handlers) &&
+        !handlers.every((handler) => typeof handler === 'string')
+      )
+    ) {
+      throw new Error('No handlers found');
+    }
+    return (handlers || []) as AoANTHandler[];
+  }
+
+  /**
+   * @param validations @type {Partial<Record<AoANTHandler, ({ ant: AoANTReadable }) => Promise<boolean>>} The validations to run on the ANT. 
+   * @returns @type {Promise<Record<AoANTHandler, { valid: boolean; error?: string }>>} The results of the validations.
+   * @example
+   * ```ts
+   * const validity = await ant.validate({
+   * validations: {
+   *  'setTicker': async ({ant}: { ant: AoANTRead }) => {
+   *   const antHandlers = await ant.getHandlers();
+            if (antHandlers.includes("setTicker")) return true;
+            throw new Error(`Handler setTicker not found`);
+          }
+    
+   * },
```

Comment:
how do you intend to use this in the arns.app? would like to see what your expectations are for integration

Reply:
![image](https://github.com/user-attachments/assets/01ba51e0-88c1-4d00-9c2b-07fd4b61b0a4)